modules/data.land/R/soilgrids_texture_extract.R

```diff
@@ -0,0 +1,129 @@
+##' soilgrids_texture_extraction function
+##' A function to extract and save three types of soil texture data in parallel for a single or group of 
+##' lat/long locations based on user-defined site location from SoilGrids250m 
+##' version 2.0 : https://soilgrids.org
+##' @title soilgrids_texture_extraction
+##' @name soilgrids_texture_extraction
+##' 
+##' @param site_info A data frame of site info containing the BETYdb site ID, 
+##' site name, latitude, and longitude, e.g. 
+##' (site_id, site_name, lat, lon)
+##' @param outdir  Provide the path to store the texture data file
+##' @param verbose Provide progress feedback to the terminal? TRUE/FALSE
+##' @return a data frame containing the soil texture data with columns "Depth", "Quantile", "Siteid", and "Value"
+##' 
+##' @export
+##' @author Qianyu Li
+##' @importFrom magrittr %>%
+
+soilgrids_texture_extraction <- function(site_info, outdir=NULL, verbose=TRUE){
+  
+  # A function to extract and save one type of soil texture data 
+  download_and_extraction <- function(base_data_url, site_info, outdir) {
+    
+    if (future::supportsMulticore()) {
+      future::plan(future::multicore)
+    } else {
+      future::plan(future::multisession)
+    }
+    
+    if (is.null(site_info)) {
+      PEcAn.logger::logger.error(
+        "No site information found. Please provide a BETY DB site list containing at least the site id and PostGIS geometry\
+    as lon and lat"
+      )
+    }
+    
+    # Prepare site info for extraction
+    internal_site_info <- data.frame(site_info$site_id, site_info$site_name, site_info$lat,site_info$lon)
```

Comment:
I think the `site_id` is enough to describe each site rather than combining `site_id` and `site_name`. Thus, you might want to just remove the `site_name` from the list to make things simpler.

Reply:
Fixed.